Wikipedia/View Controllers/WebView/WebViewController.m

```diff
@@ -915,4 +918,31 @@ - (void)editHistoryButtonPushed {
     [self presentViewController:nc animated:YES completion:nil];
 }
 
+#pragma mark - Previewing
+
+- (JSValue*)htmlElementAtLocation:(CGPoint)location {
+    return [[self.webView wmf_javascriptContext][@"getElementFromPoint"] callWithArguments:@[@(location.x), @(location.y)]];
+}
+
+- (MWKTitle*)titleForHTMLElement:(JSValue*)element {
+    if ([[[element valueForProperty:@"tagName"] toString] isEqualToString:@"A"] && [element valueForProperty:@"href"]) {
+        NSURL* url = [NSURL URLWithString:[[element valueForProperty:@"href"] toString]];
+        // Don't peek intra-page fragment links - such as references.
+        if ([url wmf_isIntraPageFragment]) {
+            return nil;
+        }
+        return [[MWKTitle alloc] initWithURL:url];
+    }
```

Comment:
There's a bug - MWKTitle's `initWithURL` returns a non-nil MWKTitle for external URLs...